builtin/game/falling.lua

```diff
@@ -369,22 +369,36 @@ core.register_entity(":__builtin:falling_node", {
 })
 
 local function convert_to_falling_node(pos, node)
-	local obj = core.add_entity(pos, "__builtin:falling_node")
-	if not obj then
-		return false
-	end
 	-- remember node level, the entities' set_node() uses this
 	node.level = core.get_node_level(pos)
 	local meta = core.get_meta(pos)
 	local metatable = meta and meta:to_table() or {}
+	local oldmeta = metatable.fields
 
 	local def = core.registered_nodes[node.name]
+
+	if def and def.on_fall then
+		local ok = def.on_fall(table.copy(pos), table.copy(node))
+		if type(ok) == "boolean" then
+			return ok
+		end
+	end
+
+	local obj = core.add_entity(pos, "__builtin:falling_node")
+	if not obj then
+		return false
+	end
 	if def and def.sounds and def.sounds.fall then
 		core.sound_play(def.sounds.fall, {pos = pos}, true)
 	end
 
 	obj:get_luaentity():set_node(node, metatable)
 	core.remove_node(pos)
+
+	if def and def.after_fall then
+		def.after_fall(pos, node, oldmeta, obj)
```

Comment:
```suggestion
		def.after_fall(vector.new(pos), node, oldmeta, obj)
```
The `pos` table can be the same as the one given to `core.spawn_falling_node` or `core.check_single_for_falling`. (It would be problematic if user provided arguments are modified by callbacks and the argument is used for something else afterwards.)